identityMatrixCheck.py

```diff
@@ -0,0 +1,72 @@
+# By Ashwath from forums
+# Given a list of lists representing a n * n matrix as input,
+# define a  procedure that returns True if the input is an identity matrix
+# and False otherwise.
+
+# An IDENTITY matrix is a square matrix in which all the elements
+# on the principal/main diagonal are 1 and all the elements outside
+# the principal diagonal are 0.
+# (A square matrix is a matrix in which the number of rows
+# is equal to the number of columns)
+
+def is_identity_matrix(matrix):
+    for i in range(len(matrix)):
+        for j in range(len(matrix[i])):
+            if len(matrix) != len(matrix[i]):
```

Comment:
If you look carefully, this statement does not use variable j. So, You can move this check before the nested loop. Everything else is good.